examples/common/metrics_foo_library/foo_library.cc

```diff
@@ -2,24 +2,43 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #ifndef ENABLE_METRICS_PREVIEW
+#  include "foo_library.h"
 #  include <chrono>
+#  include <map>
 #  include <thread>
 #  include "opentelemetry/metrics/provider.h"
 
 namespace nostd       = opentelemetry::nostd;
 namespace metrics_api = opentelemetry::metrics;
 
-void foo_library(const std::string &name)
+void foo_library::counter_example(const std::string &name)
 {
-  // Get the Meter from the MeterProvider
   auto provider                               = metrics_api::Provider::GetMeterProvider();
   nostd::shared_ptr<metrics_api::Meter> meter = provider->GetMeter(name, "1.2.0");
   auto double_counter                         = meter->CreateDoubleCounter(name);
-  double_counter->Add(28.5);
-  std::this_thread::sleep_for(std::chrono::milliseconds(500));
-  double_counter->Add(3.14);
-  std::this_thread::sleep_for(std::chrono::milliseconds(500));
-  double_counter->Add(23.5);
-  std::this_thread::sleep_for(std::chrono::milliseconds(5000));
+
+  while (true)
+  {
+    double val = (rand() % 700) + 1.1;
+    double_counter->Add(val);
+    std::this_thread::sleep_for(std::chrono::milliseconds(500));
+  }
 }
+
+void foo_library::histogram_example(const std::string &name)
+{
+  auto provider                               = metrics_api::Provider::GetMeterProvider();
+  nostd::shared_ptr<metrics_api::Meter> meter = provider->GetMeter(name, "1.2.0");
+  auto histogram_counter                      = meter->CreateDoubleHistogram(name);
+  auto context                                = opentelemetry::context::Context{};
+  std::map<std::string, std::string> labels   = {{"key", "value"}};
+  auto labelkv = opentelemetry::common::KeyValueIterableView<decltype(labels)>{labels};
+  while (true)
+  {
+    double val = (rand() % 700) + 1.1;
+    histogram_counter->Record(val, labelkv, context);
```

Comment:
May be as an enhancement later, just like `val`, we can also have `attributes` passed randomly. Define 4-5 kv-attributes, and pass one of them randomly in `Record` method.

Reply:
thanks, added random attribute.